Libraries/JavaScriptAppEngine/Initialization/InitializeJavaScriptAppEngine.js

```diff
@@ -154,6 +154,14 @@ function setUpGeolocation() {
   polyfillGlobal('geolocation', require('Geolocation'), GLOBAL.navigator);
 }
 
+function setUpProduct() {
+  let rnVersion = require('../../../package.json').version;
```

Comment:
I'm sorry but this won't work. We're trying very hard to make the size of the js file we load smaller as it has a direct impact on the time it takes to start up. Loading the entire package.json in memory just to get the version number is really bad :(

Could we punt on the version number for now and just get product there for now.

I also think that using the react-native field of package.json is going to be a more robust way to deal with detection rather than reading this global variable.